packages/backend-common/src/reading/tree/ZipArchiveResponse.ts

```diff
@@ -67,43 +68,84 @@ export class ZipArchiveResponse implements ReadTreeResponse {
 
   private shouldBeIncluded(entry: Entry): boolean {
     if (this.subPath) {
-      if (!entry.path.startsWith(this.subPath)) {
+      if (!entry.fileName.startsWith(this.subPath)) {
         return false;
       }
     }
     if (this.filter) {
-      return this.filter(this.getInnerPath(entry.path), {
-        size:
-          (entry.vars as { uncompressedSize?: number }).uncompressedSize ??
-          entry.vars.compressedSize,
+      return this.filter(this.getInnerPath(entry.fileName), {
+        size: entry.uncompressedSize,
       });
     }
     return true;
   }
 
+  private async streamToTemporaryFile(
+    stream: Readable,
+  ): Promise<{ fileName: string; cleanup: () => Promise<void> }> {
+    const tmpDir = await fs.mkdtemp(
+      platformPath.join(this.workDir, 'backstage-tmp'),
+    );
+    const tmpFile = platformPath.join(tmpDir, 'tmp.zip');
+
+    const writeStream = fs.createWriteStream(tmpFile);
+
+    return new Promise((resolve, reject) => {
+      writeStream.on('error', reject);
+      writeStream.on('finish', () =>
+        resolve({ fileName: tmpFile, cleanup: () => fs.remove(tmpFile) }),
+      );
+      stream.pipe(writeStream);
+    });
+  }
+
+  private forEveryZipEntry(
+    zip: string,
+    callback: (entry: Entry, content: Readable) => Promise<void>,
+  ): Promise<void> {
+    return new Promise((resolve, reject) => {
+      yauzl.open(zip, { lazyEntries: true }, (err, zipfile) => {
+        if (err || !zipfile) {
+          reject(err || new Error(`Failed to open zip file ${zip}`));
+          return;
+        }
+
+        zipfile.on('entry', async (entry: Entry) => {
+          if (!/\/$/.test(entry.fileName) && this.shouldBeIncluded(entry)) {
```

Comment:
Unsure about why the first condition exists :)

Reply:
It's to check that it's not a directory. I think that I dropped the comment here, I will put it back. :)